File: ruby/sample.rb

require 'json'
require_relative 'com/inmobi/monetization/ads/Banner.rb'
require_relative 'com/inmobi/monetization/ads/Native.rb'
require_relative 'com/inmobi/monetization/api/net/ErrorCode.rb'


banner = Banner.new();
r = Request.new();
property = Property.new("YOUR_PROPERTY_ID");
slot = Slot.new(15,"top");
impression = Impression.new(slot);
device = Device.new();
device.userAgent = "USER_WEBVIEW_USER_AGENT";
device.carrierIP = "USER_CARRIER_IP";
r = Request.builder(property,impression,device,nil);
#puts r.toJSON();
adsArray = nil
#adsArray = banner.loadRequest(r);
p = Hash.new();
puts p.empty?;
puts "here"
if(adsArray != nil)
	adsArray.each { |ad|
		puts ad.to_s
	}
end

native = Native.new()
r.property.propertyId = "YOUR_PROPERTY_ID";
ads = native.loadRequest(r);
if(ads != nil)
	ads.each { |ad|
		puts ad.namespace
	}
end

File: nodejs/sample.js
var Request = require('./com/inmobi/monetization/api/request/Request.js');
var Impression = require('./com/inmobi/monetization/api/request/ad/Impression.js');
var Slot = require('./com/inmobi/monetization/api/request/ad/Slot.js');
var Device = require('./com/inmobi/monetization/api/request/ad/Device.js');
var Property = require('./com/inmobi/monetization/api/request/ad/Property.js');
var BannerResponse = require('./com/inmobi/monetization/api/response/ad/BannerResponse.js');
var NativeResponse = require('./com/inmobi/monetization/api/response/ad/NativeResponse.js');
var Banner = require('./com/inmobi/monetization/ads/Banner.js');
var Interstitial = require('./com/inmobi/monetization/ads/Interstitial.js');
var Native = require('./com/inmobi/monetization/ads/Native.js');

var property = new Property("YOUR_PROPERTY_ID");
var impression = new Impression(new Slot(15,null));
var device = new Device("USER_CARRIER_IP","USER_WEBVIEW_USER_AGENT");
request = new Request(property,impression,device,null);

var banner = new Banner();
var interstitial = new Interstitial();

banner.loadRequest(request,function success(ads) {
	console.log("receive success");
		for( i in ads) {
		console.log(ads[i].adURL);
	}
},function fail(error) {
	console.log("in fail:" + error)
});

var nativeAd = new Native();
request.property.propertyId = "YOUR_PROPERTY_ID";
